File: ios/CalculatorModule.m
//
//  CalculatorModule.m
//  rnBridging
//
//  Created by Amit Kumar on 30/09/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CalculatorModule, NSObject)
 RCT_EXTERN_METHOD(add:(nonnull NSNumber *)a b:(nonnull NSNumber *)b callback:(RCTResponseSenderBlock)callback)
@end

File: ios/CalculatorModule.swift
//
//  CalculatorModule.swift
//  rnBridging
//
//  Created by Amit Kumar on 30/09/24.
//

import Foundation
import React

@objc(CalculatorModule)
class CalculatorModule: NSObject {
  
  @objc
  func add(_ a: NSNumber, b: NSNumber, callback: @escaping RCTResponseSenderBlock) {
    let sum = a.doubleValue + b.doubleValue
    callback([NSNull(), sum])
  }

  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
